### PR TITLE
Add Data Out's keyserver to our setup

### PR DESCRIPTION
**Pull request by dtitov**
It can be enabled using flag `--keyserver ega` while bootstrapping.

### CODE REVIEW

**Comment by silverdaz** [comment]
What does the data-out keyserver do that the data-in keyserver doesn't ?

File: docker/bootstrap/lega.sh [hunk 1]
```diff
@@ -231,6 +235,36 @@ cat >> ${PRIVATE}/lega.yml <<EOF
...
+      - EGA_LEGACY_PATH=
+    volumes:
+       - ./lega/pgp/ega.sec:/etc/ega/pgp/ega.sec:ro
+       - ./lega/pgp/ega.sec.pass:/etc/ega/pgp/ega.sec.pass:ro
+       - ./lega/pgp/ega2.sec:/etc/ega/pgp/ega2.sec:ro
+       - ./lega/pgp/ega2.sec.pass:/etc/ega/pgp/ega2.sec.pass:ro
+       - ./lega/pgp/ega.shared.pass:/etc/ega/pgp/ega.shared.pass:ro
+    restart: on-failure:3
+    external_links:
+      - cega-eureka:cega-eureka
```
Comment (blankdots) [inline docker/bootstrap/lega.sh]: if the profile is `no-oss` I assume it does not require `cega-eureka` ?

**Comment by blankdots** [comment]
@silverdaz 
> What does the data-out keyserver do that the data-in keyserver doesn't ?

We have been trying to align the APIs in https://github.com/EGA-archive/ega-data-api/issues/4 however the data-in keyserver (in my understanding) cannot be used by the data-out (due to some missing endpoints/functionality), thus resulting in a setup where we would have two keyservers. With this PR we are exploring the option where we would have one keyserver, in a similar idea where we would have one database https://github.com/EGA-archive/ega-data-api/issues/10

Another option is to match the functionality of the data-out keyserver in the data-in so that the implementations are interchangeable.

File: docker/bootstrap/lega.sh (hunk 1 above)
Comment (dtitov) [reply docker/bootstrap/lega.sh]: Good point. It really can be removed in this setup.

**Comment by nanjiangshu** [comment]
Hi @dtitov , I've tested with the ega keyserver and it works 👍  But it is not very straightforward to know where the flag "--keyserver ega" to be added before one look again at the Makefile. I would like to add 
```
make "ARGS=--keyserver ega" bootstrap
```
to the document for bootstrapping if the ega keyserver is going to be used.

**Comment by dtitov** [comment]
Thanks for testing, I'm glad that it works. I made a flag based on the existing flag for Inbox: `--inbox min`. But feel free to bring any changes.

**Comment by nanjiangshu** [comment]
I mean it is fine with the such flags but added explicitly in the README. I will add it. So probably you can approve and merge it?

File: docker/bootstrap/lega.sh [hunk 2]
```diff
@@ -46,10 +50,10 @@ log = debug
 #log = silent
 
 [keyserver]
-port = 8443
+port = 8080
```
Comment (blankdots) [inline docker/bootstrap/lega.sh]: flag here as well: `if [[ $KEYSERVER == 'ega' ]]; then`

**Review by blankdots — CHANGES_REQUESTED** [review]
see comments. also all Integration tests should pass even the `Makefile`-based.

**Comment by silverdaz** [comment]
I think the better approach is to update the current keyserver with the missing endpoints.

**Comment by jrambla** [comment]
Hi gentlemen,

I'm confused that we are having non-coherent Key servers, as this task was 'done' in a previous sprint, IMU...
Could someone clarify which is the situation?
Again, such changes should be discussed at the Sprint review/planning meeting. I consider switching components a major change and would like to discuss it in depth.

Best!

Obtén l'aplicació Outlook per a l'iOS<https://aka.ms/o0ukef>

________________________________
De: Frédéric Haziza <notifications@github.com>
Data d'enviament: dilluns, de setembre 17, 2018 2:38 p. m.
Per a: NBISweden/LocalEGA
A/c: Subscribed
Assumpte: Re: [NBISweden/LocalEGA] Add Data Out's keyserver to our setup (#355)


I think the better approach is to update the current keyserver with the missing endpoints.

**Comment by blankdots** [comment]
@jrambla
This PR is part of a sprint goal we set (and agreed, for sprint 36): "Start stitching a general docker-compose for LocalEGA (Data Ingestion and Data API)" and it constitutes work towards achieving that goal.

**Review by blankdots — CHANGES_REQUESTED** [review]
It would be nice to have the documentation attached to it, instead of doing it in another PR: https://github.com/NBISweden/LocalEGA/pull/356

**Review by blankdots — APPROVED** [review]
Reviewed and ega data api keyserver can be used.
With this PR we are providing the option to bootstrap with it.
Not a subject of this PR, but worth mentioning:
* Regarding the integration tests on travis, increasing sleep is not a good strategy. Waiting for the services to be up in the integration tests sounds like a more viable solution.*
* ega-data-api keyserver should have a health endpoint

**Pull request merged**